### PR TITLE
Add DotNetAddPackage alias for dotnet add package command

### LINKED ISSUE #4187: Add DotNetAddPackage alias for dotnet add package command
As per @DWIAltonaAnalytics suggestion: https://github.com/orgs/cake-build/discussions/4186#discussioncomment-6353781

Docs: https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-add-package

### PR DESCRIPTION
**Pull request by Marusyk**
Add an alias for dotnet add package command

Fixes #4187 

Please review
Thank you in advance

### CODE REVIEW

File: src/Cake.Common.Tests/Unit/Tools/DotNet/Package/Add/DotNetPackageAdderTests.cs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+using Cake.Common.Tests.Fixtures.Tools.DotNet.Package.Add;
+using Cake.Common.Tests.Fixtures.Tools.DotNet.SDKCheck;
+using Cake.Common.Tests.Fixtures.Tools.DotNet.Workload.Restore;
+using Cake.Common.Tools.DotNet;
+using Cake.Common.Tools.DotNet.Package.Add;
+using Cake.Testing;
+using Xunit;
+
+namespace Cake.Common.Tests.Unit.Tools.DotNet.Package.Add
```
Comment (Marusyk) [inline src/Cake.Common.Tests/Unit/Tools/DotNet/Package/Add/DotNetPackageAdderTests.cs:13]: What do you think, that path is correct? Then we'll able to add Package.List and Package.Remove commands

File: src/Cake.Common/Tools/DotNet/Package/Add/DotNetPackageAdder.cs [hunk 2]
```diff
@@ -0,0 +1,116 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Cake.Core;
+using Cake.Core.IO;
+using Cake.Core.Tooling;
+
+namespace Cake.Common.Tools.DotNet.Package.Add
```
Comment (Marusyk) [inline src/Cake.Common/Tools/DotNet/Package/Add/DotNetPackageAdder.cs:10]: What do you think, that path is correct? Then we'll able to add [Package.List](https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-list-package) and [Package.Remove](https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-remove-package) commands

Comment (devlead) [reply src/Cake.Common/Tools/DotNet/Package/Add/DotNetPackageAdder.cs:10]: @Marusyk Yes, that works.

**Review by devlead — CHANGES_REQUESTED** [review]
Could you please add an integration test here to
https://github.com/cake-build/cake/blob/6be615392ac27b52d979b5284993f84f175c7201/tests/integration/Cake.Common/Tools/DotNet/DotNetAliases.cake

Could use XmlPeek to verify package added, similar to 
https://github.com/cake-build/cake/blob/6be615392ac27b52d979b5284993f84f175c7201/tests/integration/Cake.Common/Tools/NuGet/NuGetAliases.cake#L94

**Comment by Marusyk** [comment]
@devlead could you help me with the failed builds. I cannot find the exact reason

**Review by devlead — APPROVED** [review]
LGTM 👍

**Comment by devlead** [comment]
@Marusyk your changes have been merged, thanks for your contribution 👍

**Pull request merged**